generic-worker tests fail without TC credentials

Lots of..
```
2020/06/08 18:20:52 Creating file /home/dustin/p/taskcluster2/workers/generic-worker/testdata/TestReclaimCancelledTask/generic-worker.config...
2020/06/08 18:20:52 Saving file /home/dustin/p/taskcluster2/workers/generic-worker/testdata/TestReclaimCancelledTask/generic-worker.config (absolute path: /home/dustin/p/taskcluster2/workers/generic-worker/testdata/TestReclaimCancelledTask/generic-worker.config)
2020/06/08 18:20:52 Loading generic-worker config file '/home/dustin/p/taskcluster2/workers/generic-worker/testdata/TestReclaimCancelledTask/generic-worker.config'...
--- FAIL: TestReclaimCancelledTask (0.00s)
    helper_broken_on_docker_test.go:50: Temporary credentials cannot be created from credentials that have an empty ClientId
    helper_test.go:551: Removing test directory /home/dustin/p/taskcluster2/workers/generic-worker/testdata/TestReclaimCancelledTask...
    helper_test.go:566: Mock HTTP services stopped
FAIL
```

--- a/changelog/HBDvjvnGTvOVUcCcCG89Zw.md
+++ b/changelog/HBDvjvnGTvOVUcCcCG89Zw.md
@@ -1,0 +1,3 @@
+audience: general
+level: silent
+---

--- a/workers/generic-worker/helper_broken_on_docker_test.go
+++ b/workers/generic-worker/helper_broken_on_docker_test.go
@@ -39,11 +39,7 @@ func CancelTask(t *testing.T) (td *tcqueue.TaskDefinitionRequest, payload Generi
 		Command:    command,
 		MaxRunTime: 300,
 	}
-	fullCreds := &tcclient.Credentials{
-		AccessToken: config.AccessToken,
-		ClientID:    config.ClientID,
-		Certificate: config.Certificate,
-	}
+	fullCreds := config.Credentials()
 	td = testTask(t)
 	tempCreds, err := fullCreds.CreateNamedTemporaryCredentials("project/taskcluster:generic-worker-tester/"+t.Name(), time.Minute, "queue:cancel-task:"+td.SchedulerID+"/"+td.TaskGroupID+"/*")
 	if err != nil {

--- a/workers/generic-worker/helper_test.go
+++ b/workers/generic-worker/helper_test.go
@@ -474,6 +474,9 @@ func GWTest(t *testing.T) *Test {
 		for _, s := range mocktc.ServiceProviders(t) {
 			s.RegisterService(r)
 		}
+		testConfig.AccessToken = "test-access-token"
+		testConfig.ClientID = "test-client-id"
+		testConfig.Certificate = ""
 	}
 
 	r.NotFoundHandler = http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {

--- a/workers/generic-worker/mocktc/queue.go
+++ b/workers/generic-worker/mocktc/queue.go
@@ -72,6 +72,10 @@ func (queue *Queue) ClaimWork(provisionerId, workerType string, payload *tcqueue
 				tcqueue.TaskClaim{
 					Task:   j.Task,
 					Status: j.Status,
+					Credentials: tcqueue.TaskCredentials{
+						ClientID:    "test-task-client-id",
+						AccessToken: "test-task-access-token",
+					},
 				},
 			)
 			if len(tasks) == int(maxTasks) {

